File: ControlRiegoAutomatico (ALPHA)/app/src/main/java/underhome/weather/com/underhome/RecyclerViewAdapter.java
package underhome.weather.com.underhome;


import android.support.v7.widget.CardView;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.daimajia.androidanimations.library.Techniques;
import com.daimajia.androidanimations.library.YoYo;

import java.util.ArrayList;
import java.util.List;

class RecyclerViewHolder extends RecyclerView.ViewHolder{

    public TextView TIPOCLIMA;
    public TextView RESUMEN;
    public TextView TEMPERATURA;
    public TextView HUMEDAD;
    CardView tarjeta;

    public RecyclerViewHolder(View itemView) {
        super(itemView);
        TIPOCLIMA = itemView.findViewById(R.id.TIPOCLIMA);
        RESUMEN = itemView.findViewById(R.id.RESUMEN);
        TEMPERATURA = itemView.findViewById(R.id.TEMPERATURA);
        HUMEDAD = itemView.findViewById(R.id.HUMEDAD);
        tarjeta = itemView.findViewById(R.id.tarjeta);
    }

}

public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerViewHolder>{

    private List<Data> listadata = new ArrayList<Data>();

    public RecyclerViewAdapter(List<Data> listadata) {
        this.listadata = listadata;
    }

    @Override
    public RecyclerViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View itemview = inflater.inflate(R.layout.item,parent,false);
        return new RecyclerViewHolder(itemview);

    }

    @Override
    public void onBindViewHolder(RecyclerViewHolder holder, int position) {
        YoYo.with(Techniques.FadeIn).playOn(holder.tarjeta);
        holder.TIPOCLIMA.setText(listadata.get(position).getA());
        holder.RESUMEN.setText(listadata.get(position).getB());
        holder.TEMPERATURA.setText(listadata.get(position).getC());
        holder.HUMEDAD.setText(listadata.get(position).getD());

    }

    @Override
    public int getItemCount() {
        return listadata.size();
    }
}

File: Menu-chip-navigation-bar/sample/src/main/java/com/ismaeldivita/chipnavigation/sample/RecyclerViewAdapter.kt
package com.ismaeldivita.chipnavigation.sample

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import com.daimajia.androidanimations.library.Techniques
import com.daimajia.androidanimations.library.YoYo
import java.util.*

class RecyclerViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    var TIPOCLIMA: TextView? = null
    var RESUMEN: TextView? = null
    var TEMPERATURA: TextView? = null
    var HUMEDAD: TextView? = null
    var tarjeta: CardView? = null

    fun RecyclerViewHolder(itemView: View) {
        TIPOCLIMA = itemView.findViewById<TextView>(R.id.TIPOCLIMA)
        RESUMEN = itemView.findViewById<TextView>(R.id.RESUMEN)
        TEMPERATURA = itemView.findViewById<TextView>(R.id.TEMPERATURA)
        HUMEDAD = itemView.findViewById<TextView>(R.id.HUMEDAD)
        tarjeta = itemView.findViewById(R.id.tarjeta)
    }
}

class RecyclerViewAdapter(listadata: MutableList<Data>) : RecyclerView.Adapter<RecyclerViewHolder>() {

    private var listadata: MutableList<Data> = ArrayList<Data>()

    fun RecyclerViewAdapter(listadata: MutableList<Data>) {
        this.listadata = listadata
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerViewHolder {
        val inflater = LayoutInflater.from(parent.context)
        val itemview = inflater.inflate(R.layout.item, parent, false)
        return RecyclerViewHolder(itemview)
    }

    override fun getItemCount(): Int {
        return listadata.size
    }

    override fun onBindViewHolder(holder: RecyclerViewHolder, position: Int) {
        YoYo.with(Techniques.FadeIn).playOn(holder.tarjeta)
        holder.TIPOCLIMA!!.text = listadata[position].getS()
        holder.RESUMEN!!.text = listadata[position].getS1()
        holder.TEMPERATURA!!.text = listadata[position].getS2()
        holder.HUMEDAD!!.text = listadata[position].getS3()
    }


}